integrations/kube-agent-updater/pkg/controller/deployment.go

```diff
@@ -91,9 +92,9 @@ func (r *DeploymentVersionUpdater) Reconcile(ctx context.Context, req ctrl.Reque
 		// Not logging the error because it provides no other information than its type.
 		log.Info("No maintenance triggered, not updating.", "currentVersion", currentVersion)
 		return requeueLater, nil
-	case trace.IsTrustError(err):
+	case errors.As(err, &trustErr):
 		// Logging as error as image verification should not fail under normal use
-		log.Error(err, "Image verification failed, not updating.")
+		log.Error(trustErr.OrigError(), "Image verification failed, not updating.")
```

Comment:
TODO: check if we can always have an origin error. I think there's a case where this is a pure trust error, not wrapping anything. Maybe it would be best to log both.

Reply:
Checked and TrustErr.OriginErr() has a sane behaviour: if its wrapped err is nil, it returns itself.